### PR TITLE
Create unit tests for `parseFlowDesc`

### PR DESCRIPTION
**Pull request by pudelkoM**
This creates an initial set of unit test for SDF related functions. Only minimal code changes to not introduce behavior changes that might break existing production setups. Those will need to be addressed over time.

### CODE REVIEW

File: pfcpiface/parse-sdf.go [hunk 1]
```diff
@@ -116,6 +105,9 @@ func parseFlowDesc(flowDesc, ueIP string) (*ipFilterRule, error) {
 	ipf := newIpFilterRule()
 
 	fields := strings.Fields(flowDesc)
+	if len(fields) < 3 {
```
Comment (pudelkoM) [inline pfcpiface/parse-sdf.go:108]: This will catch panics, so no expected breakage

File: pfcpiface/parse-sdf.go [hunk 2]
```diff
@@ -128,7 +120,7 @@ func parseFlowDesc(flowDesc, ueIP string) (*ipFilterRule, error) {
 	}
 
 	ipf.direction = fields[1]
-	ipf.proto = parseProto(fields[2])
+	ipf.proto, _ = parseL4Proto(fields[2])
```
Comment (pudelkoM) [inline pfcpiface/parse-sdf.go:123]: Ignore errors here for now

File: pfcpiface/parse-sdf.go [hunk 3]
```diff
@@ -215,18 +207,18 @@ func parseDirection(dir string) error {
...
 	switch proto {
 	case "udp":
-		return 17
+		return 17, nil
 	case "tcp":
-		return 6
+		return 6, nil
 	default:
-		return reservedProto // IANA reserved
+		return reservedProto, errBadFilterDesc
```
Comment (pudelkoM) [inline pfcpiface/parse-sdf.go:222]: Should `return 0, err`, but keep for compatibility.

**Review by osinstom — COMMENTED** [review]
Shouldn't we use `require.Equal` everywhere?

**Review by osinstom — APPROVED** [review]

**Pull request merged**